packages/firebase/src/index.ts

```diff
@@ -1,283 +1,243 @@
-import type firebase from "firebase"
-import { createHash, randomBytes } from "crypto"
-import { Adapter } from "next-auth/adapters"
-import {
-  querySnapshotToObject,
-  docSnapshotToObject,
-  stripUndefined,
-} from "./utils"
-import { Profile, Session, User } from "next-auth"
-
-interface FirebaseVerificationRequest {
-  id: string
-  identifier: string
-  token: string
-  expires: Date
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import type {
+  runTransaction,
+  collection,
+  query,
+  getDocs,
+  where,
+  limit,
+  doc,
+  getDoc,
+  addDoc,
+  updateDoc,
+  deleteDoc,
+  Firestore,
+  FirestoreDataConverter,
+} from "firebase/firestore"
+
+import type { Account } from "next-auth"
+
+import type {
+  Adapter,
+  AdapterSession,
+  AdapterUser,
+  VerificationToken,
+} from "next-auth/adapters"
+
+export const collections = {
+  Users: "users",
+  Sessions: "sessions",
+  Accounts: "accounts",
+  VerificationTokens: "verificationTokens",
+} as const
+
+export const format: FirestoreDataConverter<any> = {
+  toFirestore(object) {
+    const newObjectobject: any = {}
+    for (const key in object) {
+      const value = object[key]
+      if (value === undefined) continue
```

Comment:
👋 I've been playing around with this branch on a personal project and noticed a small issues when using the `GoogleProvider` for authentication. This line specifically checks for `undefined`, but in the case of a response from Google where `emailVerified` === `null`, reading the value below in `fromFirestore` on `value.toDate` will throw:
```
https://next-auth.js.org/errors#session_error Cannot read property 'toDate' of null TypeError: Cannot read property 'toDate' of null
```

Not sure what the best approach would be here but maybe you either want to:
1. Exclude `null` values from when writing to firebase (probably not ideal and not recommended since `null` is a valid datatype) or...
2. Ensuring `value` is defined below, before calling `toDate()`. I'll leave a suggestion below, or..
3. All of above 🤷 Again, maybe not ideal since `null` is a valid datatype but you have more context than me 😅